### ISSUE TITLE
Allow user PERL5LIB in addition to Carton's

### ISSUE THREAD

**Issue by srchulo** [perl-carton/carton#260]
Carton::CLI overwrites any `PERL5LIB `set by the user:

```
local $ENV{PERL5LIB} = "$path/lib/perl5";
```

Would it make sense to instead do something like:

```
local $ENV{PERL5LIB} = "$path/lib/perl5:$ENV{PERL5LIB}";
```

So that Carton's lib is checked first, but any libs set by the user are checked after that?

My use case for this is that I can't seem to find a way to declare local module dependencies in cpanfile, so they don't exist in local. Instead I just set my PERL5LIB to use local modules.

I could include `use lib` at the top of my scripts, but I prefer not having to write that everywhere  if possible.

**Comment by KES777** [comment]
>I can't seem to find a way to declare local module dependencies in cpanfile, 

Probably you want to configure darkpan?

Or if your modules are not installable then how you plan `carton` will install them?

**Comment by srchulo** [comment]
Yes, I think darkpan would help with that issue, that's a good suggestion.

However, I mean for this issue to be more a way to use your own `PERL5LIB` without carton overwriting it. Maybe an option could be provided to allow the existing `PERL5LIB` to not be overwritten?

**Comment by mikkoi** [comment]
I use regularly https://github.com/mikkoi/plenv-libdirs, being the developer of it.
So I need both, Carton local and additional paths to other projects which I am developing concurrently in the same system.

It is a question of preference: when an existing environment variable should be respected and when not. Plenv libdirs always respects the existing PERL5LIB with the expectation that user has deliberately set it and knows what he is doing. So it won't override.

In the case of Carton, the situation is slightly different because Carton is meant to augment the existing perl `@INC`, not replace it.

To fix this, we would need to decide if it is important to retain the current behaviour!
If yes, we will need to introduce a configuration switch (override PERL5LIB) which by default is turned on. When turned off, Carton will do: 

```perl
if( $ENV{PERL5LIB} ) {
    local $ENV{PERL5LIB} = "$carton_path:$ENV{PERL5LIB}"
} else {
    local $ENV{PERL5LIB} = "$carton_path"
}
